### ISSUE TITLE
Portal depth limit reached

### ISSUE THREAD

**Issue by timothyqiu** [godotengine/godot#51263]
### Godot version

v3.4.beta.custom_build.63047093c

### System information

macOS 11.5.1

### Issue description

I get a warning about seeing through too many portals:

```
WARNING: Portal Depth Limit reached (seeing through too many portals)
     at: trace_recursive (servers/visual/portals/portal_tracer.cpp:344)
```

But there are only 7 portals in my scene, and the portal depth limit is 16.

### Steps to reproduce

Run the MRP.

### Minimal reproduction project

[DepthLimit.zip](https://github.com/godotengine/godot/files/6935783/DepthLimit.zip)

**Comment by lawnjelly** [comment]
Ah good I was hoping someone would come up with a project for this, I had a report of it ages ago but no project. It's probably a recursion looking back through a portal, and is probably easy to fix but hard to find. Will investigate. :+1: :grinning:

**Comment by lawnjelly** [comment]
Got it. It's a peculiarity of internal rooms, where the trace can go from room 0 (outside) into room 4(the building) then out from another portal back to the outside (room 0) then back in the first portal etc... :grin:

Normally the culling planes take care of this, but in this case I think the easiest solution will be to move the trace start point during each recurse to the intersection point with the portal. I think this should solve it.

**Comment by akien-mga** [comment]
Fixed by #51274.
